controllers/gce/controller/utils.go

```diff
@@ -640,3 +673,34 @@ func (o PodsByCreationTimestamp) Less(i, j int) bool {
 	}
 	return o[i].CreationTimestamp.Before(o[j].CreationTimestamp)
 }
+
+// setInstanceGroupsAnnotation sets the instance-groups annotation with names of the given instance groups.
+func setInstanceGroupsAnnotation(existing map[string]string, igs []*compute.InstanceGroup) error {
+	type Value struct {
+		Name string
+		Zone string
+	}
+	instanceGroups := []Value{}
+	for _, ig := range igs {
+		instanceGroups = append(instanceGroups, Value{Name: ig.Name, Zone: ig.Zone})
+	}
+	jsonValue, err := json.Marshal(instanceGroups)
+	if err != nil {
+		return err
+	}
+	existing[instanceGroupsAnnotationKey] = string(jsonValue)
+	return nil
+}
+
+// uniq returns an array of unique service ports from the given array.
+func uniq(nodePorts []backends.ServicePort) []backends.ServicePort {
+	portMap := map[int64]backends.ServicePort{}
+	for _, p := range nodePorts {
+		portMap[p.Port] = p
+	}
+	nodePorts = []backends.ServicePort{}
```

Comment:
nit: either `var nodePorts []backends.ServicePort` or `nodePorts := make([]backends.ServicePort, 0 len(portMap))`

Reply:
Changed to `nodePorts = make([]backends.ServicePort, 0 len(portMap))`